congestion/rtt_stats.go

```diff
@@ -98,10 +98,10 @@ func (r *RTTStats) UpdateRTT(sendDelta, ackDelay time.Duration, now time.Time) {
 	r.updateRecentMinRTT(sendDelta, now)
 
 	// Correct for ackDelay if information received from the peer results in a
-	// positive RTT sample. Otherwise, we use the sendDelta as a reasonable
-	// measure for smoothedRTT.
+	// an RTT sample at least as large as minRTT. Otherwise, only use the
+	// sendDelta.
 	sample := sendDelta
-	if sample > ackDelay {
+	if sample-r.minRTT >= ackDelay {
```

Comment:
Is there a reason why we are ignoring the ackDelay if it would result in a value smaller than the min? Why not `max(sample - ackDelay, minRTT)`?

Reply:
That's a good question. It's specified like this in the draft: https://quicwg.github.io/base-drafts/draft-ietf-quic-recovery.html#rfc.section.3.4.5